models/Document.php

```diff
@@ -455,6 +455,21 @@ static function (array $doc) use ($oldPath, $newPath): array {
             }
             $this->clearDependentCache($additionalTags);
 
+            // if the path changed, refresh the inherited properties before updating dependencies
+            if ($differentOldPath){
+                $this->renewInheritedProperties();
+            }
+            self::updateDependendencies($this);
+
+            // refresh the inherited properties and update dependencies of each children
+            if ($differentOldPath && isset($updatedChildren) && is_array($updatedChildren)) {
```

Comment:
this only works when moving, while it should be also triggered on normal update/save 🤔 Maybe a new messenger queue for dependencies, but worried about potential drawback and race conditions.